pkg/instance/instance.go

```diff
@@ -480,17 +480,23 @@ func ExecprogCmd(execprog, executor, OS, arch, sandbox string, repeat, threaded,
 		osArg = " -os=" + OS
 	}
 	optionalArg := ""
+
+	if faultCall >= 0 {
+		optionalArg = fmt.Sprintf(" -fault_call=%v -fault_nth=%v",
```

Comment:
The intent here was that if we're passing non-zero fault injections parameters directly, then we know that we're doing - i.e. we know that we've just parsed a legacy log and invoking a legacy execprog.

In the new version, fault injection is only stored within a program itself, so it will never pass fault injection info separately from a program. At the same time, legacy logs would have fault injection info stored separately and a legacy execprog would perfectly understand this flag.

Reply:
I looked at uses of this function and it seems it's only used in automated contexts where revisions must match.
I was just thinking about the case when one builds some syzkaller tool tomorrow and takes a syzbot crash log from yesterday and it fails with obscure error message. But it does not seem to be the case here.